### PR TITLE
Various improvements to the docs

### PR DESCRIPTION
**Pull request by nrobinaubertin**
### Overview

Various improvements to the documentation:
- Remove capitalization on title and in file names
- Add a reference/integrations.md document
- Move a lot of files from tutorials to how-to
- Improve the index file (add a paragraph describing discourse, add "project & community")
- Rework the contribute document

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: docs/explanation/charm-architecture.md [hunk 1]
```diff
@@ -33,9 +33,9 @@ The server is started in HTTP mode (port `3000`) serving all the content. Alongs
 
 The workload that this container is running is defined in the [Discourse dockerfile in the charm repository](https://github.com/canonical/discourse-k8s-operator/blob/main/discourse.Dockerfile).
 
-## Docker Images
+## OCI images
 
-The image defined in the [Discourse dockerfile](https://github.com/canonical/discourse-k8s-operator/blob/main/discourse.Dockerfile) in the charm repository is published to [Charmhub](https://charmhub.io/), the official repository of charms.
+The OCI image is built using [Rockcraft](https://canonical-rockcraft.readthedocs-hosted.com) and defined in the [Rockfile](https://github.com/canonical/discourse-k8s-operator/blob/main/discourse_rock/rockcraft.yaml) in the charm repository. It is then published to [Charmhub](https://charmhub.io/), the official repository of charms.
```
Comment (arturo-seijas) [inline docs/explanation/charm-architecture.md:38]: s/repository of charms/repository for charms?

File: docs/how-to/contribute.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
+# How to contribute
+
+## Overview
+
+This document explains the processes and practices recommended for contributing
+enhancements to the Synapse operator.
```
Comment (arturo-seijas) [inline docs/how-to/contribute.md:6]: Synapse?

File: docs/index.md [hunk 3]
```diff
@@ -1,13 +1,31 @@
-For more information on Redis and PostgreSQL relations see [this section](https://charmhub.io/discourse-k8s/docs/database-relations).
+# Charmed Discourse K8s Documentation
```
Comment (arturo-seijas) [inline docs/index.md:1]: sK8s/k8s

File: docs/how-to/contribute.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+```
+
+## Build charm
+
+Build the charm in this git repository using:
+
+```shell
+charmcraft pack
+```
+For the integration tests (and also to deploy the charm locally), the synapse
```
Comment (arturo-seijas) [inline docs/how-to/contribute.md:74]: synapse?

File: docs/how-to/contribute.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+```shell
+    microk8s enable registry
+```
+
+The following commands import the images in the Docker daemon and push them into
+the registry:
+
+```shell
+    cd [project_dir]/synapse_rock && rockcraft pack rockcraft.yaml
```
Comment (arturo-seijas) [inline docs/how-to/contribute.md:85]: synapse?

File: docs/how-to/contribute.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+# Enable DEBUG logging
+juju model-config logging-config="<root>=INFO;unit=DEBUG"
+# Deploy the charm (assuming you're on amd64)
+juju deploy ./discourse-k8s_ubuntu-20.04-amd64.charm \
+  --resource discourse-image=localhost:32000/discourse:latest \
+```
+
+## Canonical Contributor Agreement
+
+Canonical welcomes contributions to the Synapse Operator. Please check out our [contributor agreement](https://ubuntu.com/legal/contributors) if you're interested in contributing to the solution.
```
Comment (arturo-seijas) [inline docs/how-to/contribute.md:104]: synapse?

File: docs/reference/integrations.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
+# Integrations
+
+### db
+
+_Interface_: pgsql
+_Supported charms_: [postgresql-k8s](https://charmhub.io/postgresql-k8s),
+[postgresql](https://charmhub.io/postgresql)
+
+Database integration is a required relation for the Synapse charm to supply
```
Comment (arturo-seijas) [inline docs/reference/integrations.md:9]: synapse?

File: docs/reference/integrations.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+### grafana-dashboard
+
+_Interface_: grafana-dashboard
+_Supported charms_: [grafana-k8s](https://charmhub.io/grafana-k8s)
+
+Grafana-dashboard relation enables quick dashboard access already tailored to
+fit the needs of operators to monitor the charm. The template for the Grafana
+dashboard for Discourse charm can be found at `/src/grafana_dashboards/discourse.json`. It was extracted from
+[matrix/synapse repository](https://github.com/matrix-org/synapse/blob/master/contrib/grafana/synapse.json). In Grafana UI, it can be
```
Comment (arturo-seijas) [inline docs/reference/integrations.md:22]: synapse?

File: docs/how-to/contribute.md (hunk 2 above)
Comment (nrobinaubertin) [reply docs/how-to/contribute.md]: Fixed

File: docs/index.md (hunk 3 above)
Comment (nrobinaubertin) [reply docs/index.md]: Fixed

File: docs/how-to/contribute.md (hunk 4 above)
Comment (nrobinaubertin) [reply docs/how-to/contribute.md]: Fixed

File: docs/how-to/contribute.md (hunk 5 above)
Comment (nrobinaubertin) [reply docs/how-to/contribute.md]: Fixed

File: docs/how-to/contribute.md (hunk 6 above)
Comment (nrobinaubertin) [reply docs/how-to/contribute.md]: Fixed

File: docs/reference/integrations.md (hunk 7 above)
Comment (nrobinaubertin) [reply docs/reference/integrations.md]: Fixed

File: docs/reference/integrations.md (hunk 8 above)
Comment (nrobinaubertin) [reply docs/reference/integrations.md]: Removed

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by merkata — APPROVED** [review]
lgtm

**Pull request merged**